dear_petition/petition/api/viewsets.py

```diff
@@ -103,6 +103,23 @@ def create(self, request):
         return resp
 
 
+class GenerateDataPetitionView(viewsets.GenericViewSet):
+
+    serializer_class = serializers.DataPetitionSerializer
+
+    def create(self, request):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        data_petition = petition.DataPetition(form_type=serializer.data["form_type"])
+        generated_petition_pdf = generate_petition_pdf(
```

Comment:
Is there any validation needed before passing in the request data to the `generate_petition_pdf()` function?

Reply:
The serializer only ensures ``form_context`` is valid JSON. Any further validation would require form-specific key/value validation, which is certainly possible, but not sure it's worth the effort at this point.

Reply:
For security, we could require some form of authentication to this view?